sdk/search/Azure.Search.Documents/tests/SearchIndexClientTests.cs

```diff
@@ -213,6 +228,7 @@ public async Task UpdateIndex()
         }
 
         [Test]
+        [SyncOnly]
```

Comment:
Why are we marking this sync only when we're using Async methods inside of it?

Reply:
Because both sync and async calls run, so there's no reason to run them twice to do exactly the same thing. I could have a condition that checks `IsAsync` and run either, but 1) that doesn't really change anything, apart from requiring another (mostly empty, granted) JSON test recording to be committed, and 2) I was following the pattern that was in place already, though both sync and async calls were running in both sync and async tests.

Reply:
Maybe I'm not following something, but doesn't the async -> sync interceptor do this for us automatically?  If we delete the `[SyncOnly]` and then the sync code path, won't it exercise both code paths without us having to write double the test code?

Reply:
Those methods are not using instrumented clients, so the interceptor has no impact. The problem is NUnit's `Assert.ThrowsAsync`. We can write our own, but I see no impetus currently. I copied the pattern that was already there, and we're getting the coverage we need.